hammer/README.md

```diff
@@ -44,3 +44,40 @@ go run ./hammer \
   --leaf_write_goal=2500 \
   --show_ui=false
 ```
+
+# Design
+
+## Objective
+
+Write a tool that can send write and read requests to a Tessera log in order to check the performance of writes and reads, and ensure that these logs are behaving correctly.
+
+## Architecture
+
+### Components
+
+Interactions with the log are performed by different implementations of worker, that are managed by separate pools:
+ - writer: adds new leaves to the tree
+ - full reader: reads all leaves from the tree, starting at 0 and fetching them all
+ - random reader: reads leaves randomly within the size of the tree
+
+All readers verify inclusion proofs against a common checkpoint, so it is cryptographically assured that they all see consistent views of the data.
+
+The number of each of these workers, and the rate at which they work is configurable (both via flags and through the UI).
+The number of workers is configured by increasing the size of the pool, which increases concurrency.
+The amount of work to be performed in a given duration is controlled by a pair of throttles: one for read operations, and one for write operations.
+
+Higher level components are:
+  - Hammer: orchestrates the work and owns the worker pools
+  - Hammer Analyser: consumes the results of the hammer and workers to determine throughput and system health
+  - UI: displays the current state of the system and allows some control over the number of workers
+
+The TUI is only intended to be used to interactively explore the capabilities of a log; actual load testing should be done headlessly.
+
+## Deployment
+
+There are 2 main modes that the hammer has been designed to run in:
+  1. Headless, which has 2 main sub-modes:
+    1. Goal-oriented: given both a timeout and a target number of entries to write, process exits successfully if enough entries can be _written_ to the log before the time expires, or otherwise an error code is returned
+    1. Infinite: keeps running until killed in order to perform long-lived performance tests
```

Comment:
For some reason, it doesn't display properly.

<img width="978" alt="image" src="https://github.com/user-attachments/assets/e776a3fc-b3a3-499b-b124-5ebbc14a6bcb">

Reply:
Fixed, cheers!